analyzers/src/SonarAnalyzer.Shared/Extensions/StatementSyntaxExtensions.cs

```diff
@@ -43,5 +47,34 @@ public static IEnumerable<StatementSyntax> GetPreviousStatements(this StatementS
                 ? previousStatements.Union(GetPreviousStatements(parentStatement))
                 : previousStatements;
         }
+
+#if CS
+        /// <summary>
+        /// Returns the statement before the statement given as input.
+        /// </summary>
+        public static StatementSyntax GetPrecedingStatement(this StatementSyntax currentStatement, ISymbol statementContainingSymbol)
+        {
+            if (statementContainingSymbol.IsTopLevelStatementEntryPoint()
+                && !currentStatement.Ancestors().Any(x => x.IsKind(SyntaxKindEx.LocalFunctionStatement)))
+            {
+                var childrenOfParentNode = currentStatement.SyntaxTree.GetCompilationUnitRoot()
+                                                                      .ChildNodes()
+                                                                      .Select(x => x.ChildNodes()
+                                                                                    .FirstOrDefault())
+                                                                      .Where(x => x != null);
+                return currentStatement.GetPrecedingStatement(childrenOfParentNode);
+            }
+            else
+            {
+                var childrenOfParentNode = currentStatement.Parent.ChildNodes();
+                return currentStatement.GetPrecedingStatement(childrenOfParentNode);
+            }
+        }
```

Comment:
As far as I can tell, it's enough to check if the parent is a `GlobalStatementSyntax` node or not. In case it is, we can check for the siblings by skipping the `GlobalStatementSyntax` nodes and using their statement.

I've tried to simplify the code a bit using the following snippet and the tests are green: 
```suggestion
        public static StatementSyntax GetPrecedingStatement(this StatementSyntax statement)
        {
            var siblings = statement.Parent is GlobalStatementSyntax
                               ? statement.SyntaxTree
                                                 .GetCompilationUnitRoot()
                                                 .ChildNodes()
                                                 .OfType<GlobalStatementSyntax>()
                                                 .Select(x => x.Statement)
                               : statement.Parent.ChildNodes();

            return statement.GetPrecedingStatement(siblings);
        }
```